### ISSUE TITLE
SCM decorators don't pass accessibility requirements

### ISSUE THREAD

**Issue by misolori** [microsoft/vscode#142809]
Not sure how this wasn't filed earlier, but based on the [1.4.1 Use of Color](https://webaim.org/articles/contrast/#sc141):

> Color is not used as the only visual means of conveying information, indicating an action, prompting a response, or distinguishing a visual element.

So the SCM indicators for added/modified along the gutter are using the same shape but with different colors:

<img width="198" alt="CleanShot 2022-02-10 at 22 09 44@2x" src="https://user-images.githubusercontent.com/35271042/153545122-aabc6e42-6681-4cb8-8886-2dddfde2f687.png">

In addition to that, the deleted icon (dark) and added/modified (light) do not pass [the minimum 3:1 color contrast ratio for non-text](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html):

<img width="402" alt="CleanShot 2022-02-11 at 08 11 46@2x" src="https://user-images.githubusercontent.com/35271042/153627248-af7872be-ceef-42ab-ad0e-edd69b5fd935.png">

<img width="89" alt="CleanShot 2022-02-11 at 08 12 35@2x" src="https://user-images.githubusercontent.com/35271042/153627278-65053fe0-28d9-4c4c-bf0c-1a03956a0430.png">

<hr>

<img width="402" alt="CleanShot 2022-02-11 at 08 14 35@2x" src="https://user-images.githubusercontent.com/35271042/153627610-55dd24b9-f048-4500-9ce1-0347026153f6.png">

<img width="402" alt="CleanShot 2022-02-11 at 08 14 46@2x" src="https://user-images.githubusercontent.com/35271042/153627643-7c27c6d3-152d-4bce-9239-7eeadc9effa3.png">

![image](https://user-images.githubusercontent.com/35271042/153627582-fd8a7003-5481-4b87-9773-9fdef6f85856.png)

cc @isidorn

**Comment by misolori** [comment]
A couple of different ideas to solve this:

## 1px line

For modified lines we could use a 1px line instead of 3px (we can still increase the line on hover)

<img width="198" src="https://user-images.githubusercontent.com/35271042/153546227-f7f4bb12-9707-48f9-967e-279a2c3bfd16.png">

## dashed line

This changes the shape type so that it's not solid

<img width="198" src="https://user-images.githubusercontent.com/35271042/153546272-e16f23c5-24f5-4005-906b-6e7cf1ee1a76.png">

## pattern line

Similar to the dashed line but with a zig zag pattern

<img width="198" src="https://user-images.githubusercontent.com/35271042/153546812-9a48cd68-22ad-43ca-923f-9bb4eb26ede6.png">

**Comment by misolori** [comment]
Here's an example if we did the pattern, also [made a little prototype](https://codepen.io/miguelsolorio/pen/JjOJJgL) to achieve the pattern with just css:

https://user-images.githubusercontent.com/35271042/153638383-5969ccb3-356b-4b2b-9f88-b325cd736bb3.mp4

**Comment by daviddossett** [comment]
+1 to the dashed or pattern line. Still has an approchable hit area compared to the 1px option and feels visually distinct.

The other potential problem with the 1px option is that it could be mistaken for a bracket pair guide.

**Comment by lychung7** [comment]
I agree with @isidorn and @daviddossett about the 1px option. I think both the dashed and pattern line helps differentiate, but I prefer the dashed line more because it can do this with more simplicity. Love what it does over hover!

**Comment by lszomoru** [comment]
+1 on the dashed line. That should be something that we can easily address.

What about?
> In addition to that, the deleted icon (dark) and added/modified (light) do not pass [the minimum 3:1 color contrast ratio for non-text](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html):

**Comment by misolori** [comment]
Here are some color changes to help pass the color contrast ratio:

## Light Theme
<img width="240" alt="CleanShot 2022-02-15 at 13 43 41@2x" src="https://user-images.githubusercontent.com/35271042/154154416-f2ce7c41-33ba-4724-ab67-0aec8a1e77e2.png">

```json
"[Default Light+]": {
    "editorGutter.modifiedBackground": "#2797DB",
    "editorGutter.addedBackground": "#51a065"
}
```

## Dark Theme

<img width="240" alt="CleanShot 2022-02-15 at 21 14 22@2x" src="https://user-images.githubusercontent.com/35271042/154200604-0ec52349-771e-4147-9917-a98fcfa9f320.png">

```json
"[Default Dark+]": {
    "editorGutter.modifiedBackground": "#0f7095",
    "editorGutter.addedBackground": "#437602",
    "editorGutter.deletedBackground": "#d30016",
}
```

**Comment by misolori** [comment]
PR https://github.com/microsoft/vscode/pull/144191 updated the SCM colors to pass the color contrast ratio, last thing needed is the adding the dashed line for modified lines

**Comment by lszomoru** [comment]
@misolori, I can take a look at that if you want. Also, should the pattern apply for the modified lines in the minimap?

**Comment by misolori** [comment]
@lszomoru yea if you wanted to try to add that, that'd be great. I tried adding it but because we are using borders and the pattern works best with background colors (from what I've seen) it wasn't that straight forward.

I think extending this to the minimap makes sense if it's doable but not a high priority since the primary area is the gutter.

**Comment by misolori** [comment]
This is now merged into main

https://user-images.githubusercontent.com/35271042/157930816-0c6f5fd1-0066-481c-ae74-1ada6291883c.mp4

**Comment by isidorn** [comment]
Looks very cool!

**Comment by i007c** [comment]
this is super annoying :/ and it doesn't have a off switch

**Comment by misolori** [comment]
@i007c please subscribe to https://github.com/microsoft/vscode/issues/146220

**Comment by DKorosec** [comment]
Sorry.. but this triggers my OCD, hard. 
Even if smooth transition is added between stripes it's still distracting. I want to concentrate on the code and this does exactly the opposite. Please add ability to revert back to default as currently all I can do is monkey patch the css on each window reload.
![image](https://user-images.githubusercontent.com/13033399/161448786-d7b086da-9ba8-4169-b78b-34d3bb3dd13d.png)

**Comment by mcanepa** [comment]
I do not like this at all!!!

but I understand that some will...

So, whoewer was in charge of making this happen: why not add an option to toggle this on/off?!

**Comment by Ark-kun** [comment]
The decorator pattern looks corrupted:

![image](https://user-images.githubusercontent.com/1829149/162543581-67405901-d67c-48ae-b24c-ac15d73a76d0.png)
